### PR TITLE
Add postgresql_grant_role resource 

### LINKED ISSUE #6: Data sources for roles
Hi,

First, well done for your provider, it's very useful!
I have a feature request, do you plan to _Data sources_ for existing roles?

Expected code could be like this:
```
data "postgresql_role" "my_admin" {
  name     = "my_admin"
}
```

And my expected behaviour is to load an existing role, and to grant new roles to it.

Thanks in advance for your feedback
Regards

Damien

### PR DESCRIPTION
**Pull request by dvdliao**
Copying from https://github.com/hashicorp/terraform-provider-postgresql/pull/189

Fix #6

### CODE REVIEW

**Comment by cyrilgdn** [comment]
@dvdliao Thanks for recreating it here, Hashicorp should have migrated the PRs & issues of the old repository but as it's still not the case we'll use this one.

I'll take a look this week-end. Could you just rebase on master?  I fixed the Travis build so it should pass now.

**Comment by dvdliao** [comment]
@cyrilgdn done. thanks!

File: postgresql/resource_postgresql_grant_role_test.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+	config := getTestConfig(t)
+	dsn := config.connStr("postgres")
+
+	dbSuffix, teardown := setupTestDatabase(t, false, true)
+	defer teardown()
+
+	_, roleName := getTestDBNames(dbSuffix)
+
+	grantedRoleName := "foo"
```
Comment (cyrilgdn) [inline postgresql/resource_postgresql_grant_role_test.go:110]: This one could also be created with a `postgresql_role` in `testAccPostgresqlGrantRoleResources` directly so you don't have to create/destroy it yourself.

**Review by cyrilgdn — CHANGES_REQUESTED** [review]
Thanks a lot for your work on this :+1: , and really sorry for the response delay.

Could you just remove the `errwrap` dependency (cf comment)  and it's good for me.
(Other comments are not mandatory)

**Comment by dvdliao** [comment]
@cyrilgdn done and thanks!

File: postgresql/resource_postgresql_grant_role_test.go [hunk 2]
```diff
@@ -58,40 +58,39 @@ func TestRevokeRoleQuery(t *testing.T) {
...
-			}),
-			expected: fmt.Sprintf("REVOKE %s FROM %s", pq.QuoteIdentifier(grantRoleName), pq.QuoteIdentifier(roleName)),
+			},
 		},
 		{
-			resource: schema.TestResourceDataRaw(t, resourcePostgreSQLGrantRole().Schema, map[string]interface{}{
+			resource: map[string]interface{}{
 				"role":              roleName,
 				"grant_role":        grantRoleName,
 				"with_admin_option": false,
```
Comment (cyrilgdn) [inline postgresql/resource_postgresql_grant_role_test.go:82]: This case is the same than the one above. You probably want to set it to true even it does not change the generated query.

**Comment by dvdliao** [comment]
@cyrilgdn done

**Review by cyrilgdn — APPROVED** [review]
Thanks a lot :+1:

**Comment by dvdliao** [comment]
hey @cyrilgdn any idea when a release will be cut? i see the milestone as completed https://github.com/cyrilgdn/terraform-provider-postgresql/milestone/2?closed=1

**Comment by cyrilgdn** [comment]
@dvdliao  FYI, this has just been released in [v1.9.0](https://github.com/cyrilgdn/terraform-provider-postgresql/releases/tag/v1.9.0).

**Pull request merged**